src/test/java/games/strategy/debug/LogReaderTest.java

```diff
@@ -0,0 +1,66 @@
+package games.strategy.debug;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.JTextArea;
+import javax.swing.SwingUtilities;
+
+import org.junit.experimental.extensions.MockitoExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+
+@ExtendWith(MockitoExtension.class)
+public class LogReaderTest {
+
+  @Mock
+  private PrintStream stream;
+  @Mock
+  private JTextArea area;
+  @Mock
+  private GenericConsole console;
+
+  @Test
+  public void testStreamSplittingArray() throws IOException, InvocationTargetException, InterruptedException {
```

Comment:
Nit: These exception specifications are just noise for test methods.  We can simplify them to `throws Exception`.